### DISCUSSION TITLE
How to apply update hook on server side of repo on github

### DISCUSSION THREAD

**Discussion by konhos** [community/community#23467] [New to GitHub]
<p>I want to discard any push which is not following particular rules. I have deployed update hook on local git repo and tested but not sure how to do the same on github repo.</p>
<p>Can someone guide?</p>

**Comment by seveas** [comment, marked as answer]
<p>It is not possible to have custom update hooks on GitHub. What people tend to do instead is to require CI checks to pass for all branches before they get merged into master, and those CI checks check all the rules.</p>

**Comment by dwhitla** [comment]
<p>Well this doesn’t help me enforce policy on pushes to feature branches. Is there some workaround</p>

**Comment by ajilani** [comment]
<p>I also would like to know how to do this, specifically for the purpose of preventing sensitive information (like key files) from being added to the repository.</p>
